src/formatting/items.rs

```diff
@@ -121,6 +121,7 @@ impl Rewrite for ast::Local {
                 mk_sp(self.pat.span.hi(), self.span.hi())
             };
 
+            let mut comment_after_assign_span = base_span;
             if let Some(offset) = context.snippet(base_span).find_uncommented("=") {
```

Comment:
At this point we already know that there is an initializer expression that's setting the value the local will be assigned to, so the assignment operator has to be in there anyway. As such this `if let..` wrapping seems unnecessary and could be removed (e.g. `let offset = context.snippet(base_span).find_uncommented("=")?;` or refactor a bit to use the `span_*` util methods on the snippet provider)

However, if you'd prefer to address the original issue with a more conservative approach that left this check in place then this would need to be refactored a bit because `base_span` would be way too big, running all the way through the end of the item.

One such approach would be to add an `else` block to this `if let..`, and utilize the original `rewrite_assign_rhs_expr` in the `else` block while moving the new `rewrite_assign_rhs_with_comments` call up inside this `if let` as the last element

Reply:
Submitted.  Took the first approach and all tests passed (used this approach as you suggested it and since a `=` is pushed to the result before that, so probably if that doesn't work it is because of a bug somewhere).

>  ... or refactor a bit to use the `span_*` util methods on the snippet provider

Didn't find a good way to refactor.  Not sure if this is because there is no such way or because I am not familiar with the "`span_*` util methods".

Reply:
> Didn't find a good way to refactor. Not sure if this is because there is no such way or because I am not familiar with the "span_* util methods".

I was referring to things like `context.snippet_provider.span_after(...` and friends but this is fine as is